src/common/state/redis.cc

```diff
@@ -779,6 +903,15 @@ void redis_task_table_test_and_update_callback(redisAsyncContext *c,
   redisReply *reply = (redisReply *) r;
   /* Parse the task from the reply. */
   Task *task = parse_and_construct_task_from_redis_reply(reply);
+  if (task == NULL) {
```

Comment:
If the task entry is not found in the table, the callback is not called. Previously, it would crash on the next line since `reply->str` was `NULL`.

It should only happen for actor tasks at the moment, but the problem is that we can't tell whether it's an actor task or not without the task entry.